src/components/post/newPost/index.js

```diff
@@ -0,0 +1,80 @@
+import React, { Fragment, useState, useEffect } from 'react';
+
+// Components
+import NewPostDisplay from './NewPostDisplay';
+
+// MUI
+import withStyles from '@material-ui/core/styles/withStyles';
+import Button from '@material-ui/core/Button';
+import style from '../../../style/style';
+
+// Icons
+import AddIcon from '@material-ui/icons/Add';
+
+// Redux
+import { connect } from 'react-redux';
+import { newUserPost } from '../../../redux/actions/dataActions';
+
+const NewPost = ({ classes = {}, UI = {}, newUserPost }) => {
+  const [open, setOpen] = useState(false);
+  const [error, setError] = useState({});
+  const [postState, setPostState] = useState({
+    body: '',
+  });
+
+  const { body } = postState;
+
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  const handleChange = event => {
+    const { name, value } = event.target;
+    setPostState({
+      [name]: value,
```

Comment:
I'm not a huge fan of this pattern. I'd rather see things set explicitly if possible.

Reply:
Can you clarify?

Reply:
the whole [name]:value thing is a little bit too arbitrary data. I could see this shortcut leading to problems that are hard to diagnose.